app/lib/core/services/database/firebase_database_interface.dart

```diff
@@ -1,124 +1,105 @@
-    var updateVehicle = await dbConnect.query('''DATABASE QUERY''');
-    dbConnect.close();
-    return updateVehicle;
-  }
-
-  Future<dynamic> updateDeliveryInfo() async {
-    dbConnect = await connectRemote();
-    var updateDelivery = await dbConnect.query('''DATABASE QUERY''');
-    dbConnect.close();
-    return updateDelivery;
-  }
+  @override
+  Future<void> newUser(FirestoreUser newUser) async =>
+      _firestore.collection('users').doc(newUser.userId).set(newUser.toJSON());
 
-  Future<dynamic> updateAckInfo() async {
-    dbConnect = await connectRemote();
-    var updateAck = await dbConnect.query('''DATABASE QUERY''');
-    dbConnect.close();
-    return updateAck;
+  @override
+  Future getRecord(User user) {
+    // TODO: implement getRecord
+    throw UnimplementedError();
   }
 
-  Future<dynamic> updateContingencyInfo() async {
-    dbConnect = await connectRemote();
-    var updateContingencyPlan = await dbConnect.query('''DATABASE QUERY''');
-    dbConnect.close();
-    return updateContingencyPlan;
-  }
+  @override
+  Future<void> setShipperInfo(String atrId, ShipperInfo shipperInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(shipperInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> setTransporterInfo(
+          String atrId, TransporterInfo transporterInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(transporterInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> saveNewAtr(InitialAtr atr) async =>
+      _firestore.collection('atr').add(atr.toJSON());
+
+  @override
+  Future<void> setAckInfo(
+          String atrId, AcknowledgementInfo acknowledgementInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(acknowledgementInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> setContingencyPlanInfo(
+          String atrId, ContingencyPlanInfo contingencyPlanInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(contingencyPlanInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> setDeliveryInfo(String atrId, DeliveryInfo deliveryInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(deliveryInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> setFwrInfo(
+          String atrId, FeedWaterRestInfo feedWaterRestInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(feedWaterRestInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> setLoadingVehicleInfo(
+          String atrId, LoadingVehicleInfo vehicleInfo) async =>
+      _firestore
+          .collection('atr')
+          .doc(atrId)
+          .set(vehicleInfo.toJSON(), SetOptions(merge: true));
+
+  @override
+  Future<void> updateAtr(InitialAtr atr) async =>
```

Comment:
- [ ] Should this take the full ATR model object eventually?

Reply:
This updates only the InitialAtr. So, userId and isComplete.  The code with SetOptions will update the objects. You can see this on the video, I used the SetOptions to update.

Reply:
I see, the name `updateATR` was misleading to me because I thought it updated the whole ATR or would eventually. I see now this is your intended way to update `isComplete`.